File: 02_CreationalDesignPatterns/01_SingletonDesignPattern/cpp/badCode.cpp
#include <iostream>

using namespace std;

class JudgeAnalytics {
private:
    int run = 0;
    int submit = 0;

public:
    void incrementRun() {
        run++;
    }

    void incrementSubmit() {
        submit++;
    }

    int getRun() {
        return run;
    }

    int getSubmit() {
        return submit;
    }
};

int main() {
    JudgeAnalytics analytics;
    analytics.incrementRun();
    analytics.incrementSubmit();
    cout << "Runs: " << analytics.getRun() << endl;
    cout << "Submissions: " << analytics.getSubmit() << endl;

    // Resetting the analytics
    JudgeAnalytics analytics2;
    cout << "After reset - Runs: " << analytics2.getRun() << endl;
    cout << "After reset - Submissions: " << analytics2.getSubmit() << endl;

    return 0;
}

File: 02_CreationalDesignPatterns/01_SingletonDesignPattern/badCode.java
class JudgeAnalytics {
    private int run = 0;
    private int submit = 0;

    public void incrementRun() {
        run++;
    }

    public void incrementSubmit() {
        submit++;
    }

    public int getRun() {
        return run;
    }

    public int getSubmit() {
        return submit;
    }


}

public class badCode {
    public static void main(String[] args) {
        JudgeAnalytics analytics = new JudgeAnalytics();
        analytics.incrementRun();
        analytics.incrementSubmit();
        System.out.println("Runs: " + analytics.getRun());
        System.out.println("Submissions: " + analytics.getSubmit());

        // Resetting the analytics
       JudgeAnalytics analytics2 = new JudgeAnalytics();
        System.out.println("After reset - Runs: " + analytics2.getRun());
        System.out.println("After reset - Submissions: " + analytics2.getSubmit());
    }
}